components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -33,40 +33,73 @@ where:
+
+[#overall_fee_calldata]
+=== Overall transaction fee with calldata
+
+This section shows the formula for determining a transaction's fee. The following sections describe how this formula was derived.
+
+The following formula describes the overall fee, stem:[F], for a transaction:
+
+[stem]
+++++
+\begin{align}
+F  = \text{gas_price}\cdot&\Bigg(\max_k v_k w_k + \\
+& + \; \text{da_calldata_cost}\left(2(n-1)+2(m-1) + \ell + 2D + 3t + \sum\limits_{i=1}^t q_i\right)\\ 
+& - \; \text{contract_update_discount}\cdot (n-1) - 240 \\
+& + \; \text{message_calldata_cost}\cdot\left(3t + \sum\limits_{i=1}^t q_i\right) \\
+& + \; \text{storage_write_cost}\cdot t \\
+& + \; \text{l2_payload_costs}\Bigg)
+\end{align}
+++++
+
+where:
+
+* stem:[$v, w, n, m, t, \ell, \text{da_calldata_cost}, \text{message_calldata_cost}, D, \text{l2_payload_costs}$] are defined in the same manner as the blob-based formula above.
 * stem:[$240$] is the gas discount for updating the sender's balance, for the derivation of this number see xref:#storage_updates[].
 * stem:[$\text{contract_update_discount}$] is 312 gas, for the derivation of this discount see xref:#storage_updates[].
-* stem:[$\text{storage_write_cost}$] is 20,000 gas, the cost of allocating a new storage cell on Ethereum.
 
 == When is the fee charged?
 
 The fee is charged atomically with the transaction execution on L2. The Starknet OS injects a transfer of the fee-related ERC-20, with an amount equal to the fee paid, the sender equal to the transaction submitter, and the sequencer as a receiver.
 
 [#fee_limitations]
-== Fee limitations
+== Transaction Fee limits
+
+[#v3_fee_limitations]
+=== v3 transactions
 
-Users can specify the maximum fee that they are willing to pay for a transaction.
+With v3 transactions, users specifiy the max amount and max price for each resource. At the time of writing, the only available resource is L1 gas. In the future, we will introduce L2 gas which will be used to price L2 work (as opposed to only charging for the proof verification in L1 gas, which is what happens today).
 
-include::partial$snippet_v3_max_price_sidebar.adoc[]
+[#deprecated_fee_limitations]
+=== Deprecated transactions (verion < 3)
 
-For transaction versions before v3 transactions, the `max_fee` field specifies the maximum fee.
+With older transactio versions, users specify the maximum fee that they are willing to pay for a transaction.
 
-The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`, but for now, StarkWare's sequencer only charges the fee required to cover the proof cost, which is potentially less than the maximum fee.
+The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`. While not enforced in the proof, the Starknet sequencer usually charges less than `max_fee`, as it charges exactly the amount given by the above fee formula.
 
```

Comment:
"exactly the amount of the formula": syscalls don't appear though, so I'm not sure about the word "exactly"